examples/react/zod/src/index.tsx

```diff
@@ -16,18 +16,28 @@ function FieldInfo({ field }: { field: FieldApi<any, any, any, any> }) {
   )
 }
 
+const formSchema = z.object({
+  firstName: z.string(),
+  lastName: z.string(),
+})
+
+type Form = z.infer<typeof formSchema>
+
 export default function App() {
   const form = useForm({
     defaultValues: {
       firstName: '',
       lastName: '',
-    },
+    } as Form,
```

Comment:
I'd probably keep the type assertion in the example for two reasons:
1) It's just one type you have to specify (and you know what type it is) instead of two, with ZodValidator being something you must know/library specific
2) On cases where the defaultValues come from an existing (typed) object you won't need any cast at all